ambry-store/src/main/java/com/github/ambry/store/HybridCompactionPolicy.java

```diff
@@ -0,0 +1,194 @@
+    CompactionPolicy selectCompactionPolicy =
+        selectCompactionPolicyAndUpdateCompactionPolicySwitchInfo(compactionPolicySwitchInfo);
+    logger.info("Current compaction policy  is : {}", selectCompactionPolicy);
+    backUpCompactionPolicyInfo(dataDir, storeId, compactionPolicySwitchInfo);
+    compactionPolicySwitchInfo.getCompactionPolicyCounter().increment();
+    return selectCompactionPolicy.getCompactionDetails(totalCapacity, usedCapacity, segmentCapacity, segmentHeaderSize,
+        logSegmentsNotInJournal, blobStoreStats, dataDir);
+  }
+
+  /**
+   * Get and Recover the CompactionPolicySwitchInfo from file.
+   * @param dataDir The directory to store the file.
+   * @param storeId id of the BlobStore
+   * @return {@link CompactionPolicySwitchInfo} gets from map or recover from file if needed.
+   */
+  private CompactionPolicySwitchInfo getAndRecoverCompactionPolicySwitchInfo(String storeId, String dataDir) {
+    if (!blobToCompactionPolicySwitchInfoMap.containsKey(storeId)) {
+      File file = new File(Paths.get(dataDir, COMPACT_POLICY_INFO_PATH).toString());
+      if (file.exists()) {
+        CompactionPolicySwitchInfo compactionPolicySwitchInfo = recoverCompactionPolicySwitchInfo(file);
+        blobToCompactionPolicySwitchInfoMap.put(storeId, compactionPolicySwitchInfo);
+      } else {
+        blobToCompactionPolicySwitchInfoMap.put(storeId,
+            new CompactionPolicySwitchInfo(new CompactionPolicyCounter(storeConfig.storeCompactionPolicySwitchPeriod),
+                INIT_COMPACT_ALL_TIME));
+      }
+    }
+    return blobToCompactionPolicySwitchInfoMap.get(storeId);
+  }
+
+  /**
+   * Recover the {@link CompactionPolicySwitchInfo} from backup file.
+   * {
+   *   "compactionPolicyCounter" : {
+   *     "storeCompactionPolicySwitchPeriod" : 3,
+   *     "counter" : 1,
+   *     "value" : 1
+   *   },
+   *   "lastCompactAllTime" : 1593463435900
+   * }
+   * @param file the backup file stores {@link CompactionPolicySwitchInfo}
+   */
+  private CompactionPolicySwitchInfo recoverCompactionPolicySwitchInfo(File file) {
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      return objectMapper.readValue(file, CompactionPolicySwitchInfo.class);
+    } catch (IOException e) {
+      logger.error("Could not deserialize file : {} into {} Object", file, CompactionPolicySwitchInfo.class.getName());
+      return new CompactionPolicySwitchInfo(new CompactionPolicyCounter(storeConfig.storeCompactionPolicySwitchPeriod),
+          INIT_COMPACT_ALL_TIME);
+    }
+  }
+
+  /**
+   * Selects which compaction policy to use for current compaction cycle.
+   * @return {@link CompactionPolicy} to use for current compaction cycle(Selects between {@link StatsBasedCompactionPolicy} and {@link CompactAllPolicy}).
+   */
+  CompactionPolicy selectCompactionPolicyAndUpdateCompactionPolicySwitchInfo(
```

Comment:
This method name is a mouthful.  Is there a downside to the caller first calling updateCompactionInfo() and then this method which could just be selectCompactionPolicy()?

Reply:
I will only update the compactionPolicySwitchInfo when we trigger the compactAllPolicy. Already re-named the method name to updateCompactionInfoWhenCompactAll() instead of updateCompactionInfo().